### ISSUE TITLE
ticket-ref: v1.4.0 false-positives on eslint-disable-next-line comments that name the rule

### ISSUE THREAD

**Issue by jay-es** [sawyerh/eslint-plugin-todo-plz#35]
## Summary

After upgrading from `1.3.2` to `1.4.0`, the `ticket-ref` rule now reports a false positive on a `// eslint-disable-next-line todo-plz/ticket-ref` comment itself, whenever that suppression comment is immediately followed by a non-conforming `TODO` comment.

## Root cause

PR #32 ("Match only standalone TODO terms") introduced `findStandaloneTermIndexes`, which treats any non `[A-Za-z0-9_]` character (including `-` and `/`) as a word boundary. Combined with PR #31's case-insensitive matching, this causes the rule to detect the literal substring `todo` inside the plugin's own rule id, `todo-plz`, as a standalone `TODO` term — because in `todo-plz/ticket-ref`, `todo` is bounded by a space before and a `-` after, both of which count as "non-word" boundaries.

As a result, a `// eslint-disable-next-line todo-plz/ticket-ref` comment is itself flagged as a `ticket-ref` violation. Since there's no suppression comment for the *disable comment's own* line, this new false positive is not suppressed and surfaces as a real lint error.

## Reproduction

`eslint.config.mjs`:
```js
import todoPlz from "eslint-plugin-todo-plz";

export default [
  {
    plugins: { "todo-plz": todoPlz },
    rules: {
      "todo-plz/ticket-ref": [
        "error",
        { commentPattern: "TODO\\: .* https://app.clickup.com/t/[0-9a-z]{9,}" },
      ],
    },
  },
];
```

`repro.js`:
```js
// eslint-disable-next-line todo-plz/ticket-ref
// TODO fix this later
const x = 1;
```

Running `eslint repro.js`:

- On `1.3.2`: no errors (exit 0) — the disable comment correctly suppresses the `TODO` violation, and the disable comment itself isn't matched as a `TODO` occurrence.
- On `1.4.0`:
  ```
  repro.js
    1:1  error  TODO comment doesn't reference a ticket number. Comment pattern: TODO\: .* https://app.clickup.com/t/[0-9a-z]{9,}  todo-plz/ticket-ref
  ```
  Note the error is reported at `1:1` — the `eslint-disable-next-line` comment's own line — not at line 2 where the actual unresolved `TODO` is.

This means any codebase that legitimately suppresses `todo-plz/ticket-ref` with a targeted `eslint-disable-next-line todo-plz/ticket-ref` comment starts failing lint after upgrading to `1.4.0`, purely because the rule id contains `todo`.

## Expected behavior

The rule should not treat text inside its own suppression-directive comments (or more generally, occurrences of `todo` that are part of an unrelated identifier like `todo-plz`) as a reportable `TODO` term. At minimum, `findStandaloneTermIndexes` should not treat `-` as a word boundary for this purpose, or the rule should ignore comments that are themselves ESLint directive comments (`eslint-disable`, `eslint-disable-next-line`, etc.).

## Environment

- `eslint-plugin-todo-plz`: 1.4.0 (regression vs 1.3.2)
- `eslint`: 9.39.5
- Introduced by #31 and #32
